src/main/com/fulcrologic/fulcro/algorithms/normalized_state_helpers.cljc

```diff
@@ -0,0 +1,427 @@
+      ;; initial state of accumulator
+      (dissoc-in state-map ident)
+      ;; a collection of values
+      idents)))
+
+
+
+(comment
+
+
+  (deep-remove-entity* normalized-state [:person/id 1])
+
+  (extract-all-idents (get-in normalized-state [:person/id 1]))
+
+  (let [item (get-in state-map ident)
+        idents (extract-all-idents item)]
+
+    (reduce
+      (fn [s i] (deep-remove-entity* s i))
+      (dissoc-in state-map ident)
+      idents))
+
+  '())
+
+
+
+;;============================================================================
+
+;; TODO
+;; - [ ] `sort-identy-by`
+;;    (sort-idents-by :entity/field vector-of-idents)
+;;    This can enable
+;;    (swap! state update-in [:entity 1 :list] sort-idents-by :list/field)
+
+
+
+;;============================================================================
+
+;; NOTE This is pretty much remove-ident*, but with the option of recursively removing the stuff
+;; NOTE assuming db-path
+(defn remove-edge*
+  ;; TODO
+  ([state-map path-to-edge])
+
+  ;; TODO needs cascade argument like remove-entity*
+  ([state-map path-to-edge cascade])
+  )
+
+(comment
+
+  (defn remove-ident*
+    "Removes an ident, if it exists, from a list of idents in app state. This
+    function is safe to use within mutations."
+    [state-map ident path-to-idents]
+    {:pre [(map? state-map)]}
+    (let [new-list (fn [old-list]
+                     (vec (filter #(not= ident %) old-list)))]
+      (update-in state-map path-to-idents new-list)))
+
+  )
+
+
+
+;============================================================================
+;; REMOVE
+(defsc A [this props]
+  {:ident         :person/id
+   :query         [:person/id :person/name]
+   :initial-state {:person/id 1 :person/name "Alice"}
+   :extra-data    42}
+  (dom/div "TODO"))
+
+
+(def ui-a (comp/factory A))
+
+
+(defn ui->props
+  "Obtain a tree of props for a UI instance from the current application state. Useful in mutations where you want
+  to denormalize an entity from the state database. `this` can often be obtained from the mutation `env` at the
+  `:component` key."
+  ([this]
+   (ui->props (comp/component->state-map this) (comp/react-type this) (comp/get-ident this)))
+  ([state-map component-class ident]
+   (fdn/db->tree (comp/get-query component-class state-map) (get-in state-map ident) state-map)))
+
+
+(comment
+
+  (comp/get-ident ui-a)
+
+  (comp/component->state-map ui-a)
+
+  (comp/component->state-map A)
+
+  (comp/react-type A)
+
+  (comp/get-ident A normalized-state)
+  (comp/get-query A)
+
+
+  '())
+
+;============================================================================
+
+;;NOTE These might belong in mutation ns???
+(defn update-caller!
+  "Runs clojure.core/update on the table entry in the state database that corresponds to the mutation caller (which
+   can be explicitly set via `:ref` when calling `transact!`).
+
+   Equivalent to `(swap! (:state env) update-in (:ref env) ...)`."
```

Comment:
Hmm, perhaps there's a clue here 

```clojure
(defn swap-entity!
  "Swap/`update-in` then entity whose ident is `ref` in the given env."
  [{:keys [state ref] :as env} & args]
  (apply swap! state update-in ref args))

```
Will be trying this out.

Reply:
These are for use within a mutation. I'm suggesting a name change because swap entity never sounded right to me. You're updating the implied caller of the mutation's state.